assembly/src/ast/ident.rs

```diff
@@ -12,11 +12,9 @@ use crate::{SourceSpan, Span, Spanned};
 pub enum IdentError {
     #[error("invalid identifier: cannot be empty")]
     Empty,
-    #[error(
-        "invalid identifier '{ident}': must contain only lowercase, ascii alphanumeric characters, or underscores"
-    )]
+    #[error("invalid identifier '{ident}': must contain only ascii graphic characters")]
```

Comment:
```suggestion
    #[error("invalid identifier '{ident}': must contain only alphanumeric or ascii graphic characters")]
```

Reply:
All ASCII alphanumeric characters are in the ASCII graphic range (i.e. alphanumerics are a subset of the range). I don't mind the rewording as a means of making it clear that alphanumerics are supported, but thought it would be redundant. Maybe there is a better way to phrase the notion of the ASCII graphic range that avoids the redundancy, but is clearer, I'm not too fixed on the specific terminology, I just didn't want to list literally every character in the range (I suppose specifying the range as hex bytes would suffice, i.e. `0x31..=0x7E`)

Reply:
> All ASCII alphanumeric characters are in the ASCII graphic range

Agreed - except we accept all *alphanumeric* characters, not only *ASCII alphanumeric* characters (e.g. `藏` is alphanumeric, but not ASCII alphanumeric). Or at least I'm saying this based on the implementation. Did you intend to use [`char::is_ascii_alphanumeric()`](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_alphanumeric)?